parsing/depend.ml

```diff
@@ -398,9 +398,10 @@ and add_module_binding bv modl =
         | _ ->  addmodule bv l; bound
       end
   | Pmod_structure s ->
-      make_node (snd (add_structure_binding bv s))
-  | _ ->
-      if !Clflags.transparent_modules then add_module bv modl; bound
+      let n = make_node (snd @@ add_structure_binding bv s) in
+      if not !Clflags.transparent_modules then add_names (collect_free n);
```

Comment:
Can you remind me when the `module M = struct ... end` path would at all depend on whether `transparent_modules` is set or not? I would naively expect that only the `module N = Path` case is affected, and that as a consequence you could just call `add_structure s` here.

Reply:
You are right that I forgot the `modtype` case.

Concerning your other question, I struggled a bit to understand why the `collect_free` is needed at all. Indeed, `collect_free` collects imported compilation units in submodules; but the compilation units imported within the current file have already been already at an earlier stage if `transparent_modules` is not set.

This `if not !Clflags.transparent_modules then add_names (collect_free n);` statement is here to record imported compilation units that comes from a map module and need to be only activated when we are not compiling dependencies for map modules . For instance,

```
(* atlas.ml *)
module A = X 
module B = Y
```

```
(*r.ml*)
include Atlas.M
```
Then, if `r.ml` is not supposed to be a map file,
```
ocamldep -map atlas.ml r.ml -modules
```
yields  `r.ml: Atlas X Y`, compared to
```
ocamldep -map atlas.ml -as-map r.ml -modules
```
that outputs simply `r.ml: Atlas` .

… this behavior may deserve some more comments in the code.